portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -162,6 +189,51 @@ private String getParameter(HttpServletRequest request, String name, String defa
     return value;
   }
 
+  /** 
+   * Returns a URL that points to the uploaded file, or null if the user 
+   * didn't upload a file.
+   */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
```

Comment:
generally good practice to do '<=' instead of a straight '==', catches more issues

Reply:
Done!